## include-and-link.dpatch by Jens Peter Secher <jps@debian.org>
##
## Use the right include directories, shared libraries and cflags.
## Use -rpath to force search in /usr/lib/neko.

Index: neko/src/tools/install.neko
===================================================================
--- neko.orig/src/tools/install.neko	2009-03-15 00:19:54.000000000 +0100
+++ neko/src/tools/install.neko	2009-03-15 00:24:38.000000000 +0100
@@ -39,11 +39,6 @@
 // LIBS DATAS
 
 libs = {
-	mod_neko => {
-		src => $array("mod_neko","cgi","../../vm/stats"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x"
-	},
 	mod_neko2 => {
 		src => $array("mod_neko","cgi","../../vm/stats"),
 		inc => $array("httpd.h","apr.h"),
@@ -56,8 +51,7 @@
 		src => $array("mysql"),
 		inc => "mysql.h",
 		incname => "MySQL 4.+"
-		lib => "libmysqlclient_r.a",
-		lparams => "-lz -lssl"
+		lparams => "-lmysqlclient"
 	},
 	regexp => {
 		src => $array("regexp"),
@@ -82,12 +76,7 @@
 		inc => switch system { "Mac" => "Carbon.h" default => "gtk/gtk.h" },
 		incname => switch system { "Mac" => "Carbon" default => "GTK+2.0" },
 		cflags => switch system { "Mac" => "" default => "`pkg-config --cflags gtk+-2.0`" },
-		lparams => switch system { "Mac" => "-framework Carbon" default => "`pkg-config --libs gtk+-2.0` -lgthread-2.0" },
-	},
-	mod_tora => {
-		src => $array("mod_tora","protocol"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x"
+		lparams => switch system { "Mac" => "-framework Carbon" default => "-lgthread-2.0" },
 	},
 	mod_tora2 => {
 		src => $array("mod_tora","protocol"),
@@ -101,15 +90,17 @@
 
 // PLATFORM
 
-cflags = "-O3 -fPIC -pthread";
+cflags = "-O3 -DEAPI -fPIC -pthread $(pkg-config --cflags apr-1)";
 cc = getenv("CC");
 if( cc == null ) cc = "gcc";
 linkcmd = switch system { "BSD" => "ld" default => cc };
 linkneko = "-lneko";
+
 linkoptions = switch system {
 	"Mac" => "-bundle -undefined dynamic_lookup -L../../bin -pthread"
-	default => "-shared -L../../bin -pthread"
+	default => "-shared -L../../bin -pthread -lneko" 
 };
+
 nekovm = switch system { "Windows" => "..\\bin\\neko" default => "../bin/neko" };
 
 if( osx_universal ) {
@@ -132,38 +123,16 @@
 search_includes = function(isap2) {
 	var inc1 = $array(
 		"/usr/include",
-		"/usr/local/include/mysql",
 		"/usr/include/mysql",
-		"/usr/local/include",
 		"/usr/include/gtk-2.0",
-		"/opt/local/include",
-		"/opt/local/include/mysql",
-		"/Developer/Headers/FlatCarbon",
 	);
 	var inc2;
-	if( isap2 )
-		inc2 = $array(
-			"/usr/local/apache2/include",
-			"/usr/include/apache2",
-			"/opt/local/apache2/include",
-			"/usr/include/apr-1",
-			"/usr/include/apr-1.0",
-			"/usr/include/apr-0",
-			"/usr/local/include/apr-1",
-			"/usr/local/include/apr-1.0",
-			"/usr/local/include/apr-0",
-			"/opt/local/include/apr-1",
-			"/opt/local/include/apr-1.0",
-			"/opt/local/include/apr-0"
-		);
-	else
-		inc2 = $array(
-			"/usr/apache/include",
-			"/usr/include/apache-1.3",
-			"/usr/include/httpd",
-			"/opt/local/include/apache",
-			"/usr/local/apache/include"
-		);
+	inc2 = $array(
+		"/usr/include/apache2",
+		"/opt/local/apache2/include",
+		"/usr/include/apr-1",
+		"/usr/include/apr-1.0",
+	);
 	return $aconcat($array(inc1,inc2));
 }
 
